client/hwlib/src/collectors/cpuid.rs

```diff
@@ -0,0 +1,115 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use anyhow::Result;
+use smbioslib;
+
+pub(super) fn get_cpuid(proc_info: &smbioslib::SMBiosProcessorInformation) -> Result<String> {
+    let cpu_id = proc_info
+        .processor_id()
+        .map(|id| format!("{:x?}", id))
+        .ok_or("Processor ID not found")
+        .unwrap();
+    Ok(cpu_id)
```

Comment:
Is this meant to panic in case of error?

I would assume this:

```suggestion
    proc_info
        .processor_id()
        .map(|id| format!("{:x?}", id))
        .ok_or("Processor ID not found")
```

Reply:
no, we should not panic in this case